datafile-handler/src/androidTest/java/com/optimizely/ab/android/datafile_handler/DatafileLoaderTest.java

```diff
@@ -110,7 +111,7 @@ public void loadFromCDNWhenNoCachedFile() throws MalformedURLException, JSONExce
 
     @Test
     public void loadWhenCacheFileExistsAndCDNNotModified() {
-        final ListeningExecutorService executor = MoreExecutors.newDirectExecutorService();
+        final ListeningExecutorService executor = MoreExecutors.listeningDecorator(Executors.newSingleThreadExecutor());
```

Comment:
unless you are calling `addListener` somewhere here, you can replace all the `listeningDecorator` stuff with just this:
```suggestion
        final ExecutorService executor = Executors.newSingleThreadExecutor();
```